elephant/online.py

```diff
@@ -0,0 +1,770 @@
+            * 0-axis --> different window
+            * 1-axis --> different pattern hash
+    n_exp : np.ndarray
+        The expected number of coincidences of each pattern within each window.
+            * 0-axis --> different window
+            * 1-axis --> different pattern hash
+    n_emp : np.ndarray
+        The empirical number of coincidences of each pattern within each
+        window.
+            * 0-axis --> different window
+            * 1-axis --> different pattern hash
+    rate_avg : np.ndarray
+        The average firing rate of each neuron of each pattern within
+        each window.
+            * 0-axis --> different window
+            * 1-axis --> different pattern hash
+            * 2-axis --> different neuron
+    indices : defaultdict
+        Dictionary contains for each trial the indices of pattern
+        within each window.
+
+    Methods
+    -------
+    get_results()
+        Returns the result dictionary with the following class attribute names
+        as keys and the corresponding attribute values as the complementary
+        value for the key: (see also Attributes section for respective key
+        descriptions)
+        * 'Js'
+        * 'indices'
+        * 'n_emp'
+        * 'n_exp'
+        * 'rate_avg'
+        * 'input_parameters'
+    update_uea(spiketrains, events)
+        Updates the entries of the result dictionary by processing the
+        new arriving 'spiketrains' and trial defining trigger 'events'.
+    reset(bw_size, trigger_events, trigger_pre_size, trigger_post_size,
+            saw_size, saw_step, n_neurons, pattern_hash, time_unit,
+            save_n_trials)
+        Resets all class attributes to their initial (default) value. It is
+        actually a re-initialization which allows parameter adjustments.
+    get_all_saved_trials()
+        Returns the last 'n'-trials which were analyzed according to the FIFO
+        strategy (first in, first out).
+
+    Returns
+    -------
+    see 'get_results()' in Methods section
+
+    Notes
+    -----
+    Common abbreviations which are used in both code and documentation:
+        bw = bin window
+        tw = trial window
+        saw = sliding analysis window
+        idw = incoming data window
+        mw = memory window
+
+    """
+
+    def __init__(self, bw_size=0.005 * pq.s, trigger_events=None,
+                 trigger_pre_size=0.5 * pq.s, trigger_post_size=0.5 * pq.s,
+                 saw_size=0.1 * pq.s, saw_step=0.005 * pq.s, n_neurons=2,
+                 pattern_hash=None, time_unit=1 * pq.s, save_n_trials=None):
+        """
+        Constructor. Initializes all attributes of the new instance.
+        """
+        # state controlling booleans for the updating algorithm
+        self.data_available_in_mv = None
+        self.waiting_for_new_trigger = True
+        self.trigger_events_left_over = True
+
+        # save constructor parameters
+        if time_unit.units != (pq.s and pq.ms):
```

Comment:
```suggestion
        if time_unit.units not in [pq.s, pq.ms]:
```
Consider the following example:
```python
import quantities as pq

time = 4*pq.s
if time.units != (pq.s and pq.ms):
    print('condition fulfilled')

output:
condition fulfilled
```
Is this intended behavior?